### PR TITLE
init

### PR DESCRIPTION
**Pull request by MaksymKos**
https://maksymkos.github.io/react_redux-list-of-posts/

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,59 +1,56 @@
...
-  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const { posts, loaded, hasError } = useAppSelector(state => state.posts);
+  const { author } = useAppSelector(state => state.author);
+  const dispatch = useAppDispatch();
+  const { selectedPost } = useAppSelector(state => state.selectedPost);
 
-  function loadUserPosts(userId: number) {
-    setLoaded(false);
+  useEffect(() => {
+    function loadUserPosts(userId: number) {
```
Comment (TarasHoliuk) [inline src/App.tsx:22]: why do you need to create function here? - you use it once

File: src/features/postsSlice.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  hasError: boolean;
+}
+
+const initialState: PostsState = {
+  posts: [],
+  loaded: false,
+  hasError: false,
+};
+
+export const postsSlice = createSlice({
```
Comment (TarasHoliuk) [inline src/features/postsSlice.ts:17]: you should create an asyncThunk (`createAsyncThunk` from `'@reduxjs/toolkit'`) and handle promise state there in the extra reducers

File: src/components/PostDetails.tsx [hunk 3]
```diff
@@ -1,92 +1,54 @@
...
-    commentsApi.getPostComments(post.id)
-      .then(setComments) // save the loaded comments
-      .catch(() => setError(true)) // show an error when something went wrong
-      .finally(() => setLoaded(true)); // hide the spinner
+    commentsApi.getPostComments(post?.id || 0)
+      .then(commentsFromServer => {
+        dispatch(commentsActions.set(commentsFromServer));
+      })
+      .catch(() => dispatch(commentsActions.setHasError(true)))
+      .finally(() => dispatch(commentsActions.setLoaded(true)));
```
Comment (TarasHoliuk) [inline src/components/PostDetails.tsx:30]: the same as with `postsSlice` you should create asyncThunk and handle all promise states in slice (in extraReducers)

**Review by TarasHoliuk — CHANGES_REQUESTED** [review]
Hey! Generally looks good but in this task you should work with asyncThunk - write it using `createAsyncThunk`.

**Review by anastasiiavorobiova — CHANGES_REQUESTED** [review]
Great work! Consider fixing the test 
Screenshot 2024-01-04 at 14 31 05

**Review by anastasiiavorobiova — CHANGES_REQUESTED** [review]
Make sure to deploy your changes

**Review by VitaliyBondarenko1982 — APPROVED** [review]
Looks good.

**Pull request closed without merging**